ext/curl/lib/hostip6.c

```diff
@@ -76,7 +76,12 @@ bool Curl_ipv6works(struct connectdata *conn)
   else {
     int ipv6_works = -1;
     /* probe to see if we have a working IPv6 stack */
+#if defined(CONFIG_NRF_CURL_INTEGRATION)
+    /* protocol with zero value not supported */
```

Comment:
Why is this combination not support, who rejects it? Socket glue layer or libmodem?

Reply:
I should have traced this time this in more details. It's have been "de facto" known issue that it isn't supported and I remember some earlier workarounds for this case but was too busy...

But now, studied more, It is failing in https://github.com/NordicPlayground/libmodem/blob/master/platform/nrf_modem_platform.c#L239
Eventually, should be fixed there, I can see elsewhere in socket() call stack, that there is some support for wildcard/protocol 0 on the call stack towards modem.

Reply:
Discussed this more. The fix is coming with in later libmodem release. But for now, this is really needed.